File: array/26.c
/*
 * 需要多用一个k来指引新数组有效位置
 */
int removeDuplicates(int* nums, int numsSize) {
    if(numsSize == 0) return 0;
    int length = numsSize;
    int temp = nums[0];
    int k = 1;
    for(int i = 1; i < numsSize; i++)
    {
        if(temp == nums[i]) length--;
        else nums[k++] = nums[i];
        temp = nums[i];
    }
    return length;
}

File: array/26.go
func removeDuplicates(nums []int) int {
    if len(nums) == 0 {
        return 0
    }
    pre := nums[0]
    count := 1
    index := 1
    
    for index < len(nums) {
        if nums[index] != pre {
            pre = nums[index]
            count++
            index++
        }else{
            nums = append(nums[:index], nums[index+1:]...)
        }
    }
    return count
}
